File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/repository/CommentRepository.java
package com.iyoa.cleanaddis.repository;

import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.iyoa.cleanaddis.model.Comment; 

public interface CommentRepository extends JpaRepository<Comment, UUID>{

	@Query(value = "SELECT * FROM comment "
			+ "WHERE post_uuid =:postUuid "
			, nativeQuery = true)
	List<Comment> getAllCommentOfPost(@Param("postUuid") UUID  postUuid);

}

File: app/src/main/java/com/iyoa/cleanaddis/repository/posting/CommentRepository.kt
package com.iyoa.cleanaddis.repository.posting

import androidx.lifecycle.LiveData
import com.iyoa.cleanaddis.data.posting.CommentDAO
import com.iyoa.cleanaddis.data.posting.CommentUUID

class CommentRepository(private val commentDAO: CommentDAO) {
    fun getAllComment():LiveData<List<CommentUUID>> = commentDAO.getAllComment()

    fun getAllCommentOfPost(postUUID:String):LiveData<List<CommentUUID>> = commentDAO.getAllCommentOfPost(postUUID)

    fun insertComment(commentUUID : CommentUUID){
        commentDAO.insertComment(commentUUID)
    }

    fun updateComment(commentUUID:CommentUUID){
        commentDAO.updateComment(commentUUID)
    }
    fun getCommentByUUID(uuid:String){
        commentDAO.getCommentByUUID(uuid)
    }
    fun deleteComment(commentUUID:CommentUUID){
        commentDAO.deleteComment(commentUUID)
    }
    fun deleteAll(){
        commentDAO.deleteAll()
    }
    fun addComments(commentUUIDS:List<CommentUUID>){
        commentDAO.addComments(commentUUIDS)
    }

}